app/src/main/java/com/github/se/eventradar/model/repository/event/IEventRepository.kt

```diff
@@ -0,0 +1,16 @@
+package com.github.se.eventradar.model.repository.event
+
+import com.github.se.eventradar.model.Resource
+import com.github.se.eventradar.model.event.Event
+
+interface IEventRepository {
+  suspend fun getEvents(): Resource<List<Event>>
+
+  suspend fun getEvent(id: String): Resource<Event?>
+
+  suspend fun addEvent(event: Event): Resource<Boolean>
+
+  suspend fun updateEvent(event: Event): Resource<Boolean>
+
+  suspend fun deleteEvent(event: Event): Resource<Boolean>
```

Comment:
Depending on how you intend to use this interface, you might not need to return `Boolean`, but `Unit` instead. For instance (e.g. `addEvent`), you could have
- the db could add the event => Resource.Success(Unit)
- the db could not add the event => Resource.Failure(error)

Reply:
_(this is kinda nitpicking)_